src/codegen/codegen_compatibility_visitor.hpp

```diff
@@ -66,6 +66,14 @@ class CodegenCompatibilityVisitor: public visitor::AstVisitor {
     /// Default CodegenCompatibilityVisitor constructor
     CodegenCompatibilityVisitor() = default;
 
+    /// CodegenCompatibilityVisitor constructor that removes the error for
+    /// the global variables if the global-to-range pass is enabled
+    explicit CodegenCompatibilityVisitor(bool enable_global_to_range) {
+        if (enable_global_to_range) {
+            unhandled_ast_types_func.erase(AstNodeType::GLOBAL_VAR);
+        }
+    }
```

Comment:
I thought that removing the error would be okay, since even if there is there is this error in the original mod file, then with the global to range pass the variables will be translated, resulting in a correct mod file representation. Otherwise, if we keep the error in every case, then to generate the code when there is this issue we would have to pass the `codegen --force` flag, which would generate the code even if there are more errors present. Do you have a better idea on how to handle this?

Reply:
Correct me if I am missing something :
* global to range transformation pass will remove problematic GLOBAL variables
* so even if we have this check in codegen_compatibility_visitor.hpp, it won't or shouldn't raise an error. Similarly, we won't need to pass `--force` for codegen because offending ones are already removed.
* my intention of keeping this check is, for some reasons if global-to-range is skipped, codegen compatibility pass will raise the error.

Reply:
You're completely right. We should just run this pass before the compatibility checker